### ISSUE TITLE
SA1115 (Parameter must follow comma) generates diagnostic when parameter list contains #pragma directives

### ISSUE THREAD

**Issue by oatkins** [DotNetAnalyzers/StyleCopAnalyzers#1886]
The following code generates a SA1115 diagnostic.

    public void Baz()
    {
        Bar(
    #pragma warning disable CS4014
            1,
    #pragma warning restore CS4014
            2);
    }

It shouldn't.

(The suppression here doesn't make sense, but it's common to use this particular suppression when intentionally not awaiting an async method.)

**Comment by sharwell** [comment]
@vweijsters @pdelvo Do we want to include this in RC 2?

**Comment by vweijsters** [comment]
I'm leaning towards including it, but there are more issues with SA1115 than the one described. For example, the following code should also not produce SA1115, but it currently does:
```csharp
        public void MyTest(int v1, int v2, int v3)
        {
        }

        public void TestMethod()
        {
            this.MyTest(
                1,
                /* comment */
                2,
                /* comment */
                3);
        }
```
The same construction should probably also be allowed for `//// comment`.

I think that these issues should also be addressed before it can be included.
